app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::VideosController < AdminsController  
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+
+    if @video.save 
+      flash[:success] = "You have successfully added the video #{@video.title}."
+      redirect_to new_admin_video_path
+    else
+      flash[:error] = "Your video could not be added. Please try again."
```

Comment:
Make sure you use `flash.now` when you use `render`. `flash` messages will survive the next page load as well, making them show twice.